testonly/hammer/hammer.go

```diff
@@ -761,6 +761,23 @@ func (s *hammerState) getSMRRev(ctx context.Context, prng *rand.Rand) error {
 	if !reflect.DeepEqual(root, smrRoot) {
 		return fmt.Errorf("get-smr-rev(@%d)=%+v, want %+v", rev, root, smrRoot)
 	}
+
+	if err := s.validateSMRMatchesWrittenContents(root); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *hammerState) validateSMRMatchesWrittenContents(root *types.MapRootV1) error {
+	mapContents := s.prevContents.PickRevision(root.Revision)
+	wantRootHash, err := mapContents.RootHash(s.cfg.MapID, s.vc.Hasher)
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(root.RootHash, wantRootHash) {
+		return fmt.Errorf("failed to check root hash: got %x, want %x", root.RootHash, wantRootHash)
```

Comment:
/failed to check/unexpected/?

Reply:
I'll update the error message in the base PR (#1872). If I had a dollar for every suggestion made on code that was moved as part of a refactoring, I'd be a rich man :-)